### PR TITLE
[Merged by Bors] - feat: Boxes in locally finite ordered rings

### PR DESCRIPTION
**Pull request by YaelDillies**
Define the sequence of "hollow boxes" indexed by natural numbers as the successive differences of the "boxes" `Icc (-n) n`.

---

This will be useful for both percolation in ℤᵈ and for bounding Eisenstein series in ℤ²

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Finset/LocallyFinite/Box.lean [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+attribute [norm_cast] toNat_ofNat
+
+lemma card_box : ∀ {n}, n ≠ 0 → (box n : Finset (ℤ × ℤ)).card = 8 * n
+  | n + 1, _ => by
+    simp_rw [Prod.card_box_succ, card_Icc, sub_neg_eq_add]
+    norm_cast
+    refine tsub_eq_of_eq_add ?_
+    zify
+    ring
+
```
Comment (CBirkbeck) [inline Mathlib/Data/Finset/LocallyFinite/Box.lean:83]: can you add the result that `box 0` has size `1`.

Comment (YaelDillies) [reply Mathlib/Data/Finset/LocallyFinite/Box.lean:83]: No need. simp already knows that: `(box 0).card = {0}.card = 1`

**Comment by CBirkbeck** [comment]
LGTM but maybe get a pro to review it as well..

File: Mathlib/Data/Finset/LocallyFinite/Box.lean [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+/-!
+# Decomposing a locally finite ordered ring into boxes
+
+This file proves that any locally finite ordered ring can be decomposed into "boxes", namely
+differences of consecutive intervals.
+
+## Implementation notes
+
+We don't need the full ring structure, only that there is an embedding `ℕ → `
```
Comment (riccardobrasca) [inline Mathlib/Data/Finset/LocallyFinite/Box.lean:20]: Does this work at least for `OrderedSemiring`? Or maybe even for additive monoid with 1 (of characteristic zero?).

**Review by riccardobrasca — APPROVED** [review]

File: Mathlib/Data/Finset/LocallyFinite/Box.lean (hunk 2 above)
Comment (YaelDillies) [reply Mathlib/Data/Finset/LocallyFinite/Box.lean:20]: No, you also need a negation. Let me clarify

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**